src/module/actor/dataModelClasses/OseDataModelCharacterEncumbrance.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * A class to handle character encumbrance.
+ */
+ export default class OseDataModelCharacterEncumbrance {
+  static encumbranceCap = 1600;
+  static encumbranceSteps = [800, 400, 200];
+  static detailedGearWeight = 80;
+
+  #encumbranceVariant;
+  #basicTreasureEncumbrance;
+  #hasAdventuringGear;
+  #weight;
+  #max;
+  #heaviestArmor;
+
+  /**
+   * 
+   * @param {number} max The max weight this character can carry
+   * @param {*} items The items this character is carrying
+   */
+  constructor(max = OseDataModelCharacterEncumbrance.encumbranceCap, items = []) {
+    const option = game.settings.get(game.system.id, "encumbranceOption");
```

Comment:
Changing system setting doesn't require a browser refresh. That being the case, the settings should probably be retrieved from `game.settings` each time their needed.

Reply:
Shower thought: I don't even think we should be retrieving this from `game.settings`, I think this should be passed in.

Reply:
My immediate reaction was to pass it in as well, but any instances of the class wouldn't know that settings have changed when computing data. It's pretty low risk, but something to consider.

Perhaps the a settings provider could be passed in the constructor arg?